floor/test/adapter/query_adapter_test.dart

```diff
@@ -282,7 +308,36 @@ void main() {
       );
 
       streamController.add(entityName);
-      streamController.add('otherEntity');
+      streamController.add(otherEntity);
+    });
+
+    test('unrelated update should not update stream', () async {
+      const otherEntity = 'otherEntity';
+      final person = Person(1, 'Frank');
+      final person2 = Person(2, 'Peter');
+      final queryResult = Future(() => [
+            <String, dynamic>{'id': person.id, 'name': person.name},
+            <String, dynamic>{'id': person2.id, 'name': person2.name},
+          ]);
+      when(mockDatabaseExecutor.rawQuery(sql)).thenAnswer((_) => queryResult);
+
+      final actual = underTest.queryListStream(sql,
+          dependencies: {entityName}, mapper: mapper);
+      //first, emit the result
+      expect(
+          actual,
+          emitsInOrder(<List<Person>>[
+            [person, person2],
+          ]));
+
+      //after emitting the first result, no other updates were made
+      expect(actual.skip(1), emitsDone);
+
+      streamController.add(otherEntity);
+      streamController.add(otherEntity);
+
+      await Future<void>.delayed(const Duration(milliseconds: 100));
+      await streamController.close();
```

Comment:
💡 We can remove this line as we close `StreamController`s in the `tearDown` block after each test.

```suggestion
```

Reply:
If I remove that, the test will fail because of a deadlock/timeout, since the `emitsInOrder` still waits for the Stream to complete, but the Stream only gets closed *after* the test succeeded.